### DISCUSSION TITLE
How much periodic task apsscheduler can handle?

### DISCUSSION THREAD

**Discussion by nitishxp** [agronholm/apscheduler#565] [Q&A]
Hi,

Just wanted to confirm before using the library and refactoring whole code. We are using postgresql and we now Google cloud scheduler (max 5000 job can be scheduled) thus we need to move away to new scheduler tech. I tried using celery since i knew about it. I tried to schedule approx 300K periodic task and the celery beat was not able to keep up with it.

My question is after the 4.0 does apsscheduler can handle our needs ? since we will be scheduling too many periodic task.

Regards

**Comment by agronholm** [comment]
I have never tried APScheduler with that many jobs. I guess you need to test it yourselves. It also depends greatly on the expected rate of the schedules and the workload they impose on the workers. The upcoming APScheduler 4.0 should make this easier by allowing multiple concurrent, independent schedulers and workers to run against the same data store.

**Reply by nitishxp** [reply]
@agronholm The best part of our workload is that they just invoke a cloud function which is pretty fast. so the worker will not be blocked that much over a single task. 
Is there any tentative line for 4.0 to be released?

**Reply by agronholm** [reply]
I'm hoping to get a pre-release out this year and a final release some time next year. That's all I can say.

**Reply by agronholm** [reply]
You should probably do some load testing with 3.x to see what the potential bottlenecks are. Note that 4.0 is a complete redesign and while a migration path will be offered at some point, it is not a top priority for the pre-releases.

**Reply by nitishxp** [reply]
Sounds good :)
